atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CombiningHostLocationSupplier.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.pool;
+
+import java.util.Optional;
+
+final class CombiningHostLocationSupplier implements HostLocationSupplier {
+    private final HostLocationSupplier overrideLocation;
+    private final HostLocationSupplier locationSupplier;
+
+    CombiningHostLocationSupplier(
+            HostLocationSupplier overrideLocationSupplier,
+            HostLocationSupplier locationSupplier) {
+        this.overrideLocation = overrideLocationSupplier;
+        this.locationSupplier = locationSupplier;
+    }
+
+    @Override
+    public Optional<HostLocation> get() {
+        Optional<HostLocation> location = overrideLocation.get();
+        if (location.isPresent()) {
+            return location;
+        }
+
+        return locationSupplier.get();
```

Comment:
non-actionable: ... and this is where I complain about not having Java 9 `Optional#or` which does exactly what you need

Reply:
😢